miniapps/nurbs/nurbs_patch_ex2.cpp

```diff
@@ -0,0 +1,317 @@
+//                    boundary --->| material | material |<--- boundary
+//                    attribute 1  |    1     |    2     |     attribute 2
+//                    (fixed)      +----------+----------+     (pull down)
+//
+//               This example is a specialization of ex2 which demonstrates
+//               patch-wise partial assembly on NURBS meshes.
+
+#include "mfem.hpp"
+#include <fstream>
+#include <iostream>
+
+using namespace std;
+using namespace mfem;
+
+int main(int argc, char *argv[])
+{
+   // 1. Parse command-line options.
+   const char *mesh_file = "../../data/beam-hex-nurbs.mesh";
+   bool pa = false;
+   bool patchAssembly = false;
+   int ref_levels = 0;
+   int nurbs_degree_increase = 0;  // Elevate the NURBS mesh degree by this
+   int spline_integration_type = 0;
+   int preconditioner = 0;
+   int visport = 19916;
+   bool csv_info = 0;
+   bool visualization = 1;
+
+   OptionsParser args(argc, argv);
+   args.AddOption(&mesh_file, "-m", "--mesh",
+                  "Mesh file to use.");
+   args.AddOption(&pa, "-pa", "--partial-assembly", "-no-pa",
+                  "--no-partial-assembly", "Enable Partial Assembly.");
+   args.AddOption(&patchAssembly, "-patcha", "--patch-assembly", "-no-patcha",
+                  "--no-patch-assembly", "Enable patch-wise assembly.");
+   args.AddOption(&ref_levels, "-ref", "--refine",
+                  "Number of uniform mesh refinements.");
+   args.AddOption(&nurbs_degree_increase, "-incdeg", "--nurbs-degree-increase",
+                  "Elevate NURBS mesh degree by this amount.");
+   args.AddOption(&spline_integration_type, "-int", "--integration-type",
+                  "Integration rule type: 0 - full Gaussian, "
+                  "1 - reduced Gaussian");
+   args.AddOption(&preconditioner, "-pc", "--preconditioner",
+                  "Preconditioner: 0 - none, 1 - diagonal");
+   args.AddOption(&csv_info, "-csv", "--csv-info", "-no-csv",
+                  "--no-csv-info",
+                  "Enable or disable dump of info into csv.");
+   args.AddOption(&visualization, "-vis", "--visualization", "-no-vis",
+                  "--no-visualization",
+                  "Enable or disable GLVis visualization.");
+   args.Parse();
+
+   // Print & verify options
+   args.PrintOptions(cout);
+   MFEM_VERIFY(!(pa && !patchAssembly), "Patch assembly must be used with -pa");
+   MFEM_VERIFY(spline_integration_type >= 0 && spline_integration_type < 2,
+               "Spline integration type must be 0 or 1 for this example");
+
+   // 2. Read the mesh from the given mesh file.
+   Mesh mesh(mesh_file, 1, 1);
+   const int dim = mesh.Dimension();
+   const bool isNURBS = mesh.NURBSext;
+
+   // Verify mesh is valid for this problem
+   MFEM_VERIFY(isNURBS, "Example is for NURBS meshes");
+   MFEM_VERIFY(mesh.GetNodes(), "NURBS mesh must have nodes");
+   if (mesh.bdr_attributes.Max() < 2)
+   {
+      cout << "\nInput mesh should have at least two boundary"
```

Comment:
Comparing to `ex2`, should there also be a check for two material attributes? Shouldn't the example terminate in this case?